pkg/kubelet/flannel_helper.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	utildbus "k8s.io/kubernetes/pkg/util/dbus"
+	utilexec "k8s.io/kubernetes/pkg/util/exec"
+	utiliptables "k8s.io/kubernetes/pkg/util/iptables"
+
+	"github.com/golang/glog"
+)
+
+// TODO: Move all this to a network plugin.
+const (
+	dockerOptsFile    = "/etc/default/docker"
+	flannelSubnetKey  = "FLANNEL_SUBNET"
+	flannelNetworkKey = "FLANNEL_NETWORK"
+	flannelMtuKey     = "FLANNEL_MTU"
+	dockerOptsKey     = "DOCKER_OPTS"
+	flannelSubnetFile = "/var/run/flannel/subnet.env"
```

Comment:
flannel actually writes it out to `/run/flannel/subnet.env` by default. They're often symlinked but not always.